HW04/Ivan_Papushoi/array_my_methods_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require_relative 'ivan_papushoi_hw04.rb'
+require 'rspec/autorun'
+
+RSpec.describe Array do
+  describe '#my_select' do
+    context 'with array and odd? method' do
+      let(:array) { [1, 2, 3, 4, 5, 6] }
+      let(:term) { ->(x) { x.odd? } }
+      let(:expectation) { [1, 3, 5] }
+
+      it { expect(array.my_select(&term)).to eq(expectation) }
+    end
+
+    context 'when selecting negatives from random array' do
+      let(:array) { described_class.new(20) { rand(-100..100) } }
+      let(:term) { ->(x) { x.negative? } }
+
+      it { expect(array.my_select(&term)).to eq(array.select(&term)) }
+    end
+  end
+
+  describe '#my_each' do
+    let(:expectation) { 'value 1 value 2 value 3 ' }
+    let(:term) { ->(x) { print "value #{x} " } }
+    let(:array) { [1, 2, 3] }
+
+    context 'when method printing each value in stdout' do
```

Comment:
We're using context if we want to separate specs based on conditions ( `empty/not empty array, with/without block`) There's no need to create context if you don't have any specific conditions.

Reply:
You can move your variables under context or just remove it
```suggestion
    context 'when method printing each value in stdout' do
      let(:expectation) { 'value 1 value 2 value 3 ' }
      let(:term) { ->(x) { print "value #{x} " } }
      let(:array) { [1, 2, 3] }
```

Reply:
Moved variables under context